petastorm/spark/spark_dataset_converter.py

```diff
@@ -1,94 +1,111 @@
         :param cache_file_path: The path to store the cache files.
         """
         self.cache_file_path = cache_file_path
-        self.dataset = None
 
-    def make_tf_dataset(self) -> tf_dataset_context_manager:
+    def __len__(self):
+        return 0  # todo
+
+    def make_tf_dataset(self):
         reader = make_batch_reader("file://" + self.cache_file_path)
-        self.dataset = make_petastorm_dataset(reader)
-        return tf_dataset_context_manager(self)
+        return tf_dataset_context_manager(reader)
 
-    def close(self):
+    def delete(self):
         """
-        Todo: delete cache files
+        Delete cache files at self.cache_file_path.
         :return:
         """
-        pass
+        shutil.rmtree(self.cache_file_path, ignore_errors=True)
+
+
+class tf_dataset_context_manager:
+
+    def __init__(self, reader: Reader):
+        self.reader = reader
+        self.dataset = make_petastorm_dataset(reader)
+
+    def __enter__(self) -> tf.data.Dataset:
+        return self.dataset
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.reader.stop()
+        self.reader.join()
+
+
+def _get_uuid():
+    """
+    Generate a UUID from a host ID, sequence number, and the current time.
+    :return: a string of UUID.
+    """
+    return str(uuid.uuid1())
 
 
-def _cache_df_or_retrieve_cache_path(df: DataFrame, dir_path: str) -> str:
+def _cache_df_or_retrieve_cache_path(df: DataFrame, cache_dir: str) -> str:
     """
     Check whether the df is cached.
     If so, return the existing cache file path.
-    If not, cache the df into the configured cache dir in parquet format and return the cache file path.
-    :param df: SparkDataFrame
-    :param dir_path: the directory for the saved parquet file, could be local, hdfs, dbfs, ...
-    :return: the path of the saved parquet file
+    If not, cache the df into the cache_dir in parquet format and return the cache file path.
+    :param df:        A :class:`DataFrame` object.
+    :param cache_dir: The directory for the saved parquet file, could be local, hdfs, dbfs, ...
+    :return:          The path of the saved parquet file.
     """
-    # Todo: add cache management
+    uuid_str = _get_uuid()
+    save_to_dir = os.path.join(cache_dir, uuid_str)
     df.write.mode("overwrite") \
         .option("parquet.block.size", 1024 * 1024) \
-        .parquet(CACHE_DIR)
+        .parquet(save_to_dir)
 
     # remove _xxx files, which will break `pyarrow.parquet` loading
-    underscore_files = [f for f in os.listdir(dir_path) if f.startswith("_")]
+    underscore_files = [f for f in os.listdir(save_to_dir) if f.startswith("_")]
     for f in underscore_files:
-        os.remove(os.path.join(dir_path, f))
-    return CACHE_DIR
+        os.remove(os.path.join(save_to_dir, f))
+    return save_to_dir
 
 
-def make_spark_converter(df: DataFrame) -> SparkDatasetConverter:
+def make_spark_converter(df: DataFrame, cache_dir=None) -> SparkDatasetConverter:
```

Comment:
Remove return type `-> SparkDatasetConverter`

Reply:
Why? Is it because we don't want to use python typing syntax to keep consistent with other modules of petastorm? If so, I will remove all the typing hints.

Reply:
I removed them.